src/ui/download-repo-dialog.cpp

```diff
@@ -188,12 +119,16 @@ bool DownloadRepoDialog::validateInputs()
                              QMessageBox::Ok);
         return false;
     }
-    QDir dir(mDirectory->text());
-    if (!dir.exists()) {
-        QMessageBox::warning(this, getBrand(),
-                             tr("The folder does not exist"),
-                             QMessageBox::Ok);
-        return false;
+    sync_with_existing_ = false;
+    // exist and is non-empty?
+    if (isNonEmptyDirectory(mDirectory->text())) {
+        sync_with_existing_ = true;
+        int ret = QMessageBox::question(
```

Comment:
It can reduce the code here from 5 lines to 2 lines, and the name `yerOrNoBox` shows the intent of the code more clearly.